RecoBTag/CTagging/interface/CharmTagger.h

```diff
@@ -0,0 +1,63 @@
+#ifndef RecoBTag_CTagging_CharmTagger_h
+#define RecoBTag_CTagging_CharmTagger_h
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CommonTools/Utils/interface/TMVAEvaluator.h"
+#include "RecoBTau/JetTagComputer/interface/JetTagComputer.h"
+#include <mutex>
+#include "FWCore/Utilities/interface/ESInputTag.h"
+#include "RecoBTag/SecondaryVertex/interface/CombinedSVSoftLeptonComputer.h"
+#include "DataFormats/BTauReco/interface/TaggingVariable.h"
+#include "RecoBTau/JetTagComputer/interface/JetTagComputerRecord.h"
+
+//DEBUGGING
+#include "TFile.h"
+#include "TNtuple.h"
+#include <memory>
+#include <map>
+
+/** \class CharmTagger
+ *  \author M. Verzetti, U. Rochester, N.Y.
+ *  copied from ElectronTagger.h
+ */
+
+class CharmTagger : public JetTagComputer {
+public:
+  /// explicit ctor 
+	CharmTagger(const edm::ParameterSet & );
+	~CharmTagger();//{}
+  virtual float discriminator(const TagInfoHelper & tagInfo) const override;
+	virtual void initialize(const JetTagComputerRecord & record) override;
+	
+	typedef std::vector<edm::ParameterSet> vpset;
+	
+	struct MVAVar {
+		std::string name;
+		reco::btau::TaggingVariableName id;
+		size_t index;
+		bool has_index;
+		float default_value;
+	};
+
+private:
+  mutable std::mutex mutex_;
+	[[cms::thread_guard("mutex_")]] std::unique_ptr<TMVAEvaluator> mvaID_;
```

Comment:
TMVAEvaluator ***does*** use the GBRForest implementati under the hood, but still allows for using plain TMVA.  I left the option to choose from the configuration. Given that I thought that it was safer to leave the mutex. For consistency I could remove the option and fix to GBRForest and remove the mutex.

BTW, for my own learning, what does the syntax ```  [[cms::thread_guard("mutex_")]]  ``` mean? I looked it up on Google but I did not quite understand the effect.